lib/index.js

```diff
@@ -2850,12 +2858,16 @@ addToplevelMethod('pipeline', function (/*through...*/) {
         });
     });
 
+    wrapper.writable = true;
+
     wrapper.write = function (x) {
         return start.write(x);
     };
 
     wrapper.end = function () {
-        return start.end();
+        var result = start.end();
+        wrapper._onEnd();
```

Comment:
That's right. `this.writable`.

`_onEnd` is called when a stream is destroyed (via the `destroy()`) method or when it *emits* a `nil`. This is not the same thing as when a `nil` is *written* to it. Specifically,

```javascript
var s = _();
s.end();
// Here, s.writable should be false. However, the stream hasn't emitted any data yet,
// so it hasn't been destroyed. `_onEnd` has not been called.
s.pull((err, x) => {
    // Here, `_onEnd` has been called.
});
```

Looking at the code again, since `destroy()` calls `end()`, we should be able to remove the `writable = false` in `_onEnd`. Sorry for giving the wrong advice earlier.